internal/distributedtracing/distributedtracing.go

```diff
@@ -0,0 +1,277 @@
+package distributedtracing
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/open-policy-agent/opa/config"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/util"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	defaultEnable               = false
+	defaultAddress              = "localhost:4317"
+	defaultServiceName          = "opa"
+	defaultSampleRatePercentage = int(100)
+	defaultEncyrptionScheme     = "off"
+	defaultEncryptionSkipVerify = false
+	defaultTLSCertFile          = ""
```

Comment:
I'm not sure I understand why we'd need these. We could also _not_ set them "as defaults" in `validateAndInjectDefaults`...